### PR TITLE
Report cache hits/misses

### PR DESCRIPTION
**Pull request by bbockelm**
This PR tracks whether there was a cache hit or miss based on the `Age` header in the server response.  The goal is to provide some indication to users in the hold message as to why their transfer failed -- was it a slow origin or a slow cache?  (The assumption is if it was a cache hit and a slow transfer then the cache is at fault.)

Also in this PR:
- Minor cleanup of the debug log messages when doing transfers.  Less repetition; do not print caches that we won't ever use.
- Fix the pre-transfer queries to selected cache hosts; particularly, put caches that respond with an error to the back of the list.
- Send correct header for PROPFIND request.
- Fix a small "airplane mode" issue causing tests to fail when there's no internet connectivity.

### CODE REVIEW

File: client/handle_http.go [hunk 1]
```diff
@@ -1796,7 +1883,7 @@ func downloadHTTP(ctx context.Context, te *TransferEngine, callback TransferCall
 	req = req.WithContext(ctx)
 
 	// Test the transfer speed every 5 seconds
```
Comment (jhiemstrawisc) [inline client/handle_http.go:1885]: Can you update this comment

File: client/handle_http.go [hunk 2]
```diff
@@ -2144,14 +2246,14 @@ func uploadObject(transfer *transferFile) (transferResult TransferResults, err e
...
 	var lastKnownWritten int64
-	t := time.NewTicker(20 * time.Second)
-	defer t.Stop()
 	uploadStart := time.Now()
 
 	go runPut(request, responseChan, errorChan)
 	var lastError error = nil
 
 	tickerDuration := 100 * time.Millisecond
+	stoppedTransferTimeout := time.Duration(param.Client_StoppedTransferTimeout.GetInt()) * time.Second
```
Comment (jhiemstrawisc) [inline client/handle_http.go:2255]: Can this be simplified to use param's `GetDuration()`?

File: client/handle_http_test.go [hunk 3]
```diff
@@ -851,3 +861,203 @@ func TestSearchJobAd(t *testing.T) {
 		assert.Equal(t, "12345", jobId)
 	})
 }
+
+// Test error messages when a 504 Gateway Timeout occurs
+func TestGatewayTimeout(t *testing.T) {
+	viper.Reset()
+	viper.Set("Logging.Level", "debug")
+	config.InitConfig()
```
Comment (jhiemstrawisc) [inline client/handle_http_test.go:869]: And here.

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

File: config/config.go [hunk 4]
```diff
@@ -1438,10 +1438,6 @@ func InitClient() error {
 
 	upper_prefix := GetPreferredPrefix()
 
-	viper.SetDefault("Client.StoppedTransferTimeout", 100)
-	viper.SetDefault("Client.SlowTransferRampupTime", 100)
-	viper.SetDefault("Client.SlowTransferWindow", 30)
```
[the remark this answers is not in the document]
Comment (bbockelm) [reply config/config.go:1443]: The code above breaks in a different direction: these parameters need to be set for the local cache (which calls `InitServer`) but only have defaults in `InitClient`.

By moving this to `defaults.yaml`, it's available to both client and server.

The problem Ismael was hitting is he didn't invoke `InitClient` at all -- I believe the best approach there is to have the client API throw an error when it's used without `InitClient()` being called.

**Review by jhiemstrawisc — APPROVED** [review]
LGTM

**Pull request merged**